Products/ZenModel/migrate/collectorDaemonsImportZproxy.py

```diff
@@ -0,0 +1,43 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2015, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import Migrate
+import servicemigration as sm
+sm.require("1.0.0")
+
+
+class CollectorDaemonsImportZproxy(Migrate.Step):
+    "Import zproxy in all collector daemons."
+
+    version = Migrate.Version(5,1,0)
+
+    def cutover(self, dmd):
+        ctx = sm.ServiceContext()
+
+        # Get all services tagged "collector" and "daemon".
+        collectorDaemons = filter(lambda s: all(x in s.tags for x in ["collector", "daemon"]), ctx.services)
+
+        # Create the endpoint we're importing.
+        endpoint = sm.Endpoint(
+            name = "zproxy",
+            purpose = "import",
+            application = "zproxy",
+            portnumber = 8080,
+            protocol = "tcp",
+            addressConfig = sm.AddressConfig(0, "")
+        )
+
+        # Add the endpoint to each daemon.
+        for daemon in collectorDaemons:
+            daemon.endpoints.append(endpoint)
+
+        # Commit our changes.
+        ctx.commit()
```

Comment:
Whatever you guys do with the migration scripts whenever possible if we can get some idea that things are still processing and not hung or looping...that is super important.  if there is a big change happening things can get hung up (like if we make a mistake and have a hub still connected to zodb).  the zup actually does a good job with it but the individual zenpacks dont always...from and end user perspective, the worst scenerio is you have planned downtime of x hours and you are running straces to see if stuff is hung because you are creaping up on your change backout time.  i know this isnt probably directly relevant to what you are doing here ..